posthog/migrations/0834_delete_invalid_personalapikeys.py

```diff
@@ -0,0 +1,50 @@
+# Generated by Django 4.2.22 on 2025-08-28 22:37
+import logging
+
+from django.db import migrations, models
+
+logger = logging.getLogger(__name__)
+
+
+def purge_null_secure_values(apps, schema_editor):
+    PersonalAPIKey = apps.get_model("posthog", "PersonalAPIKey")
+
+    base_query = PersonalAPIKey.objects.filter(secure_value__isnull=True)
+
+    # get ids that will be deleted
+    ids = []
+    for pk in base_query.values_list("id", flat=True).iterator(chunk_size=1000):
+        ids.append(pk)
+
+    ids_str = ", ".join(str(i) for i in ids)
+    logger.debug(f"Deleting {len(ids)} rows: {ids_str}")
+
+    deleted, _ = base_query.delete()
+    logger.debug(f"Deleted {deleted} rows.")
```

Comment:
We perform an extra query so that we can log the deleted ids, just in case something goes horrible wrong and we need to restore from backup.